src/goBuild.ts

```diff
@@ -70,7 +70,7 @@ export function goBuild(fileUri: vscode.Uri, goConfig: vscode.WorkspaceConfigura
 	}
 
 	const buildEnv = Object.assign({}, getToolsEnvVars());
-	const tmpPath = path.normalize(path.join(os.tmpdir(), 'go-code-check.' + getUserNameHash()));
+	const tmpPath = getTempFile('go-code-check.' + getUserNameHash());

```

Comment:
We can skip the use of user name hash here which was introduced to fix #1829
The current solution of using `fs.mkdtempSync` should fix #1829 as well.